src/main/java/org/jboss/tm/TransactionLocal.java

```diff
@@ -26,296 +26,257 @@
-      if(value == null)
-      {
-         // get the initial value
-         value = initialValue();
-
-         // if value is null replace it with the null value standin
-         if(value == null)
-         {
+public class TransactionLocal {
+
+    /**
+     * To simplify null values handling in the preloaded data pool we use this value instead of
+     * 'null'
+     */
+    private static final Object NULL_VALUE = new Object();
+
+    /**
+     * The transaction manager is maintained by the system and manges the assocation of transaction
+     * to threads.
+     */
+    protected final TransactionManager transactionManager;
+
+    /** The delegate */
+    protected TransactionLocalDelegate delegate;
+
+    /**
+     * Creates a thread local variable.
+     *
+     * @throws IllegalStateException if there is no system transaction manager
+     */
+    public TransactionLocal() {
+        transactionManager = TransactionManagerLocator.locateTransactionManager();
+        initDelegate();
+    }
+
+    /**
+     * Creates a transaction local variable. Using the given transaction manager
+     *
+     * @param tm the transaction manager
+     */
+    public TransactionLocal(TransactionManager tm) {
+        if (tm == null) throw new IllegalArgumentException("Null transaction manager");
+        this.transactionManager = tm;
+        initDelegate();
+    }
+
+    /**
+     * Lock the TransactionLocal using the current transaction
+     *
+     * <p>
+     *
+     * <p>WARN: The current implemention just "locks the transactions"
+     *
+     * @throws IllegalStateException if the transaction is not active
+     * @throws InterruptedException if the thread is interrupted
+     */
+    public void lock() throws InterruptedException {
+        lock(getTransaction());
+    }
+
+    /**
+     * Lock the TransactionLocal using the provided transaction
+     *
+     * <p>
+     *
+     * <p>WARN: The current implemention just "locks the transactions"
+     *
+     * @param transaction the transaction
+     * @throws IllegalStateException if the transaction is not active
+     * @throws InterruptedException if the thread is interrupted
+     */
+    public void lock(Transaction transaction) throws InterruptedException {
+        // ignore when there is no transaction
+        if (transaction == null) return;
+
+        delegate.lock(this, transaction);
+    }
+
+    /** Unlock the TransactionLocal using the current transaction */
+    public void unlock() {
+        unlock(getTransaction());
+    }
+
+    /**
+     * Unlock the ThreadLocal using the provided transaction
+     *
+     * @param transaction the transaction
+     */
+    public void unlock(Transaction transaction) {
+        // ignore when there is no transaction
+        if (transaction == null) return;
+
+        delegate.unlock(this, transaction);
+    }
+
+    /**
+     * Returns the initial value for this thransaction local. This method will be called once per
```

Comment:
Did checkstyle suggest this line length or did you use a reformatting tool, if so which one? I only ask because the original line lengths looked ok.